### PR TITLE
Download at least recipe in conan download command

### PR DESCRIPTION
**Pull request by danimtb**
- [x] Refer to the issue that supports this Pull Request: #2261
- [x] If the issue has missing info, explain the purpose/use case/pain/need that covers this Pull Request.
- [x] I've read the [Contributing guide](https://raw.githubusercontent.com/conan-io/conan/develop/.github/CONTRIBUTING.md).
- [x] I've followed the PEP8 style guides for Python code.

### CODE REVIEW

**Comment by danimtb** [comment]
Shall I include complete tests for all ``$ conan download`` command options?

File: conans/test/command/download_test.py [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+
+class DownloadTest(unittest.TestCase):
+
+    def download_reference_without_packages_test(self):
+        client = TestClient()
+        conanfile = """from conans import ConanFile
+class Pkg(ConanFile):
+    settings = "os"
+"""
+        client.save({"conanfile.py": conanfile})
```
Comment (lasote) [inline conans/test/command/download_test.py]: useless?

File: conans/test/command/download_test.py [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+
+        server = TestServer()
+        servers = {"default": server}
+
+        client = TestClient(servers=servers, users={"default": [("lasote", "mypass")]})
+
+        files = cpp_hello_conan_files()
```
Comment (lasote) [inline conans/test/command/download_test.py]: I think you don't need real cpp packages, with the conanfile you are good.

Comment (memsharded) [reply conans/test/command/download_test.py]: Yes, exactly, this test can be simplified.

File: conans/test/command/download_test.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+        client.run("export . lasote/stable")
+
+        ref = ConanFileReference.loads("pkg/0.1@lasote/stable")
+        self.assertTrue(os.path.exists(client.paths.conanfile(ref)))
+
+        client.run("upload pkg/0.1@lasote/stable")
+        client.run("remove pkg/0.1@lasote/stable -f")
+        self.assertFalse(os.path.exists(client.paths.export(ref)))
+
+        client.run("download pkg/0.1@lasote/stable", ignore_error=True)
```
Comment (memsharded) [inline conans/test/command/download_test.py]: If you are using ``ignore_error`` please capture the return of client and check that it is True (error = True)

File: conans/test/command/download_test.py [hunk 4]
```diff
@@ -0,0 +1,79 @@
...
+
+        client.run("upload pkg/0.1@lasote/stable")
+        client.run("upload %s -p %s" % (str(ref), package_ref.package_id))
+        client.run("remove pkg/0.1@lasote/stable -f")
+        self.assertFalse(os.path.exists(client.paths.export(ref)))
+
+        client.run("download pkg/0.1@lasote/stable")
+
+        # Check not 'No remote binary packages found' warning
+        self.assertNotIn("WARN: No remote binary packages found in remote", client.out)
```
Comment (memsharded) [inline conans/test/command/download_test.py]: This is the test that is failing in CI. Why shouldn't be packages? You just created and uploaded a binary too.

File: conans/test/command/download_test.py (hunk 3 above)
Comment (danimtb) [reply conans/test/command/download_test.py]: ``ignore_error=True`` was not necessary indded. Thanks

File: conans/test/command/download_test.py (hunk 4 above)
Comment (danimtb) [reply conans/test/command/download_test.py]: There should be a package there but conan dowload seems to fail dowloading it or I migth be doing something wrong here in the test...

**Pull request merged**